apps/frontend/core-dapp/src/features/portfolio/PortfolioValue.tsx

```diff
@@ -0,0 +1,98 @@
+import { Flex, Icon, spacingIncrement } from 'prepo-ui'
+import styled from 'styled-components'
+import { observer } from 'mobx-react-lite'
+import prepoLogoBgDark from './prepo-logo-bg-dark.svg'
+import prepoLogoBgLight from './prepo-logo-bg-light.svg'
+import { useRootStore } from '../../context/RootStoreProvider'
+import { numberFormatter } from '../../utils/numberFormatter'
+import Skeleton from '../../components/Skeleton'
+
+const { toUsd } = numberFormatter
+
+const Container = styled(Flex)`
+  background: url('${({ theme }): string =>
+      (theme.isDarkMode ? prepoLogoBgDark : prepoLogoBgLight).src}')
+    no-repeat center;
+  background-size: cover;
+  border-radius: ${spacingIncrement(16)};
+  box-shadow: ${({ theme }): string => theme.shadow.prepo};
+  height: fit-content;
+`
+const Label = styled.span`
+  color: ${({ theme }): string => theme.color.neutral3};
+  font-size: ${spacingIncrement(14)};
+  line-height: ${spacingIncrement(14)};
+`
+
+const SmallLabel = styled(Label)`
+  font-size: ${spacingIncrement(12)};
+`
+
+const Value = styled.span`
+  color: ${({ theme }): string => theme.color.secondary};
+  font-size: ${spacingIncrement(16)};
+  font-weight: ${({ theme }): number => theme.fontWeight.semiBold};
+  line-height: ${spacingIncrement(16)};
+`
+
+const BigValue = styled(Value)`
+  font-size: ${spacingIncrement(32)};
+  line-height: ${spacingIncrement(34)};
+`
+
+const ActionLink = styled.a`
+  align-items: center;
+  display: flex;
+  font-size: ${spacingIncrement(12)};
+  font-weight: ${({ theme }): number => theme.fontWeight.semiBold};
+  line-height: ${spacingIncrement(12)};
+`
+
+const PortfolioValue: React.FC = () => {
+  const { portfolioStore, preCTTokenStore } = useRootStore()
+  const { portfolioValue, tradingPositionsValue } = portfolioStore
+  const { tokenBalanceFormat } = preCTTokenStore
+
+  return (
+    <Container p={16} flexDirection="column" alignItems="start" justifyContent="start" gap={24}>
+      <Flex flexDirection="column" alignItems="start" gap={4}>
+        <SmallLabel>Portfolio Value</SmallLabel>
+        {portfolioValue === undefined ? (
+          <Skeleton width={120} height={34} />
+        ) : (
+          <BigValue>{toUsd(portfolioValue)}</BigValue>
```

Comment:
good point on > 99M exceeding the box, can use `compactNumber` util function here to make it show `123.45M`

the `compactNumber` has maximumFractionDigits option set as 2, could you help set minimumFractionDigits to 2 as well so it's always 2 decimals

```suggestion
          <BigValue>${compactNumber(+portfolioValue)}</BigValue>
```

Reply:
Good catch, I also set it to currency = USD since every place where we used it we prepended a dollar sign. Removed that hardcoded dollar sign and renamed the function to `compactUsd`.

I don't like that now it shows $1.1K instead of $1100,00. I think thousands are compact enough to display them fully. If that's fine by you, I'll create a ticket and add more logic and tests in another PR to solve this.

Reply:
ohh we should keep $ out of it because there will definitely be use case where we need to show compacted value without dollar sign (e.g. when we migrate sales page later, we will need to show `12.34K PPO staked`

As for $1.1k vs $1100, I think we should let it compact when above 999.99 because some number can be fatter and wont fit on smaller screen (e.g. 2888)

Reply:
> ohh we should keep $ out of it because there will definitely be use case where we need to show compacted value without dollar sign (e.g. when we migrate sales page later, we will need to show 12.34K PPO staked

I did this to avoid the `$-10K` issue, maybe we can have `compactNumber(value, { showUsdSign: true })`?

Reply:
good point, yeap `showUsdSign` solution sounds good to me, it should default to false or undefined tho, and only show dollar sign when explicitly specified, since not every use case uses the dollar sign.